interp/notation.ml

```diff
@@ -1872,14 +1872,16 @@ let collect_notations stack =
       | SingleNotation ntn ->
           if List.mem_f notation_eq ntn knownntn then (all,knownntn)
           else
+          try
             let { not_interp  = (_, r); not_location = (_, df) } =
               NotationMap.find ntn (find_scope default_scope).notations in
             let all' = match all with
               | (s,lonelyntn)::rest when String.equal s default_scope ->
                   (s,(df,r)::lonelyntn)::rest
               | _ ->
                   (default_scope,[df,r])::all in
-            (all',ntn::knownntn))
+            (all',ntn::knownntn)
+          with Not_found -> (* e.g. if only printing *) (all,knownntn))
```

Comment:
How about using `find_opt` ?

Reply:
Yes, this could be an alternative. Here, the initial code was not supposed to fail before only printing was introduced. So, in the original code, a `find_opt` would anyway have required an `assert false` in the `None` case. Afaicj, this is a case where dependent typing would have been needed to detect the bug.